sharepa/analysis.py

```diff
@@ -16,6 +16,33 @@ def bucket_to_dataframe(name, bucket):
     ).T
 
 
+def convert_key(x):
+    x.key = x.key_as_string
+    return x
+
+
+def aggregation_to_dataframe(name,agg,filter=None):
+    '''A function that turns elasticsearch response with/without aggregation into a dataframe
+
+        :param name: The name of the returned dataframe col
+        :type name: str
+        :param bucket: a bucket from elasticsearch results
+        :type bucket: list[dict]
+        :returns: pandas.DataFrame
+    '''
+    if hasattr(agg, 'buckets'):
+        if filter is not None:
+            expanded_agg = []
+            for bucket in agg.buckets:
+                expanded_agg.append(bucket_to_dataframe(bucket.key, map(convert_key, getattr(bucket,filter).buckets)))
```

Comment:
Not all aggregations will have the ```key_as_string``` key. Might be better to make dataframes that are just indexed by row number, and make each key in the bucket it's own column, perhaps namespaced by the name. So like:

 | dates.key | dates.key_as_string | dates.\<other_thing>
--------|----------|----------|----------
0 | key1 | key_as_string1 | \<other_thing>1